### DISCUSSION TITLE
画面遷移を伴うピンの削除(情報の受け渡し)

### DISCUSSION THREAD

**Discussion by Keita0920** [lit-iphone-course/lit-iphone-course#73] [Question-original] [answered]
(1) 実現したいこと
マップに表示したピンを削除したい。プロセスとしてはマップに表示された写真をタップすると削除ボタンがついているモーダルが出てくるのでそれを押すとピンが消えるようにしたい。

(2) 実現するために自分が試した内容とその結果
写真を設定したピンを立てるのにmapView.addAnnotation("カスタムクラスのピンの変数")としているので消す時もmapView.removeAnnotation("カスタムクラスのピンの変数")をすれば消えると思ったがnilになってしまい、変数の受け渡しが上手くいかない。

(3) 調べた内容や資料
1.https://qiita.com/nomunomu0504/items/2ee54efdb666a2eed5e4

(4) 原因だと思われそうな箇所(ソースコードをコピペして貼る)
遷移元のクラスでピンをタップしたら処理が行われる関数があったのでその中でviewを受け取って(nilになってしまう)、
`//ピンが選択されたら
    func mapView(_ mapView: MKMapView, didSelect view: MKAnnotationView) {
        performSegue(withIdentifier: "toHalfModal", sender: nil)
        print(view)
        annotationView=view
        }`

ここでキャストして渡そうとしている。
`override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
            let next = segue.destination
            if let sheet = next.sheetPresentationController {
                sheet.detents = [.medium()]

                //モーダル出現後も親ビュー操作可能にする
                sheet.largestUndimmedDetentIdentifier = .medium
                // 角丸の半径を変更する
                sheet.preferredCornerRadius = 40.0
            }
        if segue.identifier == "toHalfModal" {
            let secondView = segue.destination as! HalfModalViewController
            secondView.mapView = mapView
            secondView.annotationPin=(annotationView as! any MKAnnotation)
        }
    }`

(5) GitHubのリンク
https://github.com/Keita0920/LA_RePlace

**Comment by yukiny0811** [comment, marked as answer]
zoomで話してます
```
やることとしては、今annotationの中にobjectが入るようになったから、削除ボタンを押した時にそのobjectを削除すればOK！
```
